examples/msgpack/msgpack_entity.go

```diff
@@ -0,0 +1,42 @@
+package restPack
+
+import (
+	restful "github.com/emicklei/go-restful"
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+const MIME_MSGPACK = "application/x-msgpack" // Accept or Content-Type used in Consumes() and/or Produces()
+
+// NewEntityAccessorMPack returns a new EntityReaderWriter for accessing MessagePack content.
+// This package is not initialized with such an accessor using the MIME_MSGPACK contentType.
+func NewEntityAccessorMsgPack(contentType string) restful.EntityReaderWriter {
+	return entityMsgPackAccess{ContentType: contentType}
+}
+
+// entityOctetAccess is a EntityReaderWriter for Octet encoding
+type entityMsgPackAccess struct {
+	// This is used for setting the Content-Type header when writing
+	ContentType string
```

Comment:
I think you get ContentType json because you are skipping a part of go-restful in your test.

Perhaps you should write a simple example program, similar to the hello-world, that create a WebService, sets both Consumes and Produces to msgpack and show how the handler works.